src/features/MergeNetworks/models/Impl/CreateMergedNetworkWithView.ts

```diff
@@ -0,0 +1,102 @@
+    putTablesToDb,
+    putVisualStyleToDb,
+    putNetworkViewToDb,
+    putNetworkSummaryToDb,
+} from '../../../../store/persist/db'
+import { NetworkWithView } from '../../../../utils/cx-utils';
+import { mergeNetwork } from './MergeNetwork';
+import { IdType } from '../../../../models/IdType';
+import { NetworkRecord, NetworktoMerge } from '../DataInterfaceForMerge';
+import VisualStyleFn, { VisualStyle } from '../../../../models/VisualStyleModel';
+import ViewModelFn, { NetworkView } from '../../../../models/ViewModel';
+import { deepClone } from '../../utils/helper-functions';
+import { NetworkAttributes } from '../../../../models/NetworkModel';
+import { Column } from '../../../../models/TableModel/Column';
+import { MatchingTable } from '../MatchingTable';
+import { getMatchingTableRows, getAttributeMapping } from './MatchingTableImpl';
+import { Visibility } from '../../../../models/NetworkSummaryModel/Visibility';
+
+export const createMergedNetworkWithView = async (fromNetworks: IdType[], toNetworkId: IdType, networkRecords: Record<IdType, NetworkRecord>,
+    nodeAttributeMapping: MatchingTable, edgeAttributeMapping: MatchingTable, networkAttributeMapping: MatchingTable,
+    matchingAttribute: Record<IdType, Column>, visualStyle: VisualStyle): Promise<NetworkWithView> => {
+    if (fromNetworks.length < 1) {
+        throw new Error("No networks to merge");
+    }
+    if (getMatchingTableRows(nodeAttributeMapping).length < 2) {
+        throw new Error("Attributes Length should be greater than 1")
+    }
+    for (const netId of fromNetworks) {
+        if (!networkRecords[netId]) {
+            throw new Error(`Network with id ${netId} not found`);
+        }
+        if (!getAttributeMapping(nodeAttributeMapping, netId)) {
+            throw new Error(`Node attribute mapping for network ${netId} not found`);
+        }
+        if (!matchingAttribute[netId]) {
+            throw new Error(`Matching attribute for network ${netId} not found`);
+        }
+    }
+    const newNetworkName = 'Merged Network'
+    const mergedNetwork: NetworkRecord = mergeNetwork(fromNetworks, toNetworkId, networkRecords,
+        nodeAttributeMapping, edgeAttributeMapping, networkAttributeMapping, matchingAttribute)
+
+    const baseNetSummary = 'Merged Network' //todo fecth from db
+    const newNetworkDescription = 'Merged Network'
+    // todo: merge network attributes also
+    const networkAttributes: NetworkAttributes = {
+        id: toNetworkId,
+        attributes: {},
+    }
+
+    const newNetwork = mergedNetwork.network
+    const newNodeTable = mergedNetwork.nodeTable
+    const newEdgeTable = mergedNetwork.edgeTable
+
+    // Initialize new visual style and network view model
+    const newVisualStyle: VisualStyle = deepClone(visualStyle) || VisualStyleFn.createVisualStyle();
+    const newNetworkView: NetworkView = ViewModelFn.createViewModel(newNetwork)
+
+    await putNetworkSummaryToDb({
+        ownerUUID: toNetworkId,
```

Comment:
There are changes to the network summary in my table data loader branch.  

Specifically, a flag called `isNdex` needs to be set to false here.

I believe from past discussions it seemed reasonable that the result of merging networks produces a local network.

Reply:
Okay, I will add this property